demo/primefaces-users-portlet/src/main/java/com/liferay/faces/demos/bean/UsersBackingBean.java

```diff
@@ -121,9 +141,7 @@ public void save(ActionEvent actionEvent) {
 
 		try {
 
-			if (!userLocalServiceTracker.isEmpty()) {
```

Comment:
@juangon It would seem that UserLocalServiceTracker already has the `addingService` and `removedService` capability built-into it since it extends org.osgi.util.tracker.ServiceTracker ([see source code here](http://grepcode.com/file/repo1.maven.org/maven2/org.apache.felix/org.apache.felix.framework/3.0.4/org/osgi/util/tracker/ServiceTracker.java)) Would you agree?

Reply:
Exactly. It would be great to reuse these code by injecting a bean that would have all these OSGI code into these beans. Do you want me to follow that approach and resent?

Reply:
@juangon Well what I was trying to say was the deleted UserLocalServiceTracker.java class that extends org.osgi.util.tracker.ServiceTracker and inherits the `addingService` and `removedService`. So I'm a little unclear as to why UserLocalServiceTracker.java was not already adequate.

Reply:
@ngriffin7a problem is we need a reference here that would be updated whenever needed. So, we would need to add the userLocalService reference in those classes, and have an API to return them by the classes that needs those references. So, having a custom ServiceTracker without adding these references that would be updated is not enough.

Reply:
So, UserLocalServiceTracker would need a private UserLocalService userLocalService. Then update this reference in addingService etc, then add a method to return this. And use this method from rest of classes.

Reply:
@juangon I think that storing an instance of `UserLocalServiceTracker` in the bean is OK, because we always check `if (!userLocalServiceTracker.isEmpty())` before calling
`userLocalServiceTracker.getService()` which always gives us a fresh reference. In other words, we *always* make sure we get the updated instance of the service before we try and use it.

Reply:
@ngriffin7a if you agree I can implement this code in UserLocalServiceTracker etc.

Reply:
I don't think so @ngriffin7a . Following examples in Liferay and checking ServiceTracker code (http://grepcode.com/file/repo1.maven.org/maven2/org.apache.felix/org.apache.felix.framework/3.0.4/org/osgi/util/tracker/ServiceTracker.java#ServiceTracker.getService%28%29) you can see service is cached and not updated when service is being reloaded.

Reply:
@juangon I guess I'm still not understanding how the existing UserLocalServiceTracker (prior to any PRs) manifests FACES-3241 as a problem. Seems to me like everything is fine? If I am wrong then maybe you could send me a follow-up PR that shows how to fix UserLocalServiceTracker. Thanks.